### PR TITLE
Fix mozilla/thimble.webmaker.org#1184 - Do not inject remix code for readonly projects

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

**Comment by gideonthomas** [comment]
@Pomax nice catch :)
Could you r when you get a chance?

File: api/classes/publisher.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ function uriSafe(path) {
  */
 function upload(path, buffer, remixMetadata) {
   var mimeType = mime.lookup(path);
-  if (mimeType === 'text/html') {
+  if (mimeType === 'text/html' && remixMetadata) {
```
Comment (Pomax) [inline api/classes/publisher.js]: it's better to rely on `remixMetaData.readonly` here (1/2)

File: api/classes/publisher.js [hunk 2]
```diff
@@ -178,6 +178,10 @@ function updateProjectDetails() {
 function setRemixDataForPublishedProject() {
   var self = this;
 
+  if (self.project.readonly) {
```
Comment (Pomax) [inline api/classes/publisher.js]: (2/2) and then not do this, but build the real meta data object, and assign the `readonly` flag as `true`.

**Comment by gideonthomas** [comment]
@Pomax fixed! Does it look good now?

**Comment by Pomax** [comment]
r+

**Pull request merged**